### PR TITLE
Allowing for git branches to be used that resemble SHAs

### PR DESCRIPTION
**Pull request by mikefero**

### CODE REVIEW

File: ccmlib/repository.py [hunk 1]
```diff
@@ -113,10 +113,32 @@ def clone_development(git_repo, version, verbose=False):
...
 
+                # determine if the request is for a branch
+                is_branch = False
+                try:
+                    branch_listing = subprocess.check_output(['git', 'branch', '--all'], cwd=target_dir)
+                    branch_listing = branch_listing.replace('remotes/origin/', '')
+                    for branch in branch_listing.splitlines(True):
+                        if git_branch == branch.strip():
+                            is_branch = True
+                except subprocess.CalledProcessError, cpe:
```
Comment (ptnapoleon) [inline ccmlib/repository.py]: Is this also valid python 3 syntax?

File: ccmlib/repository.py [hunk 2]
```diff
@@ -113,10 +113,32 @@ def clone_development(git_repo, version, verbose=False):
                 else:
                     subprocess.call(['git', 'clone', local_git_cache, target_dir], cwd=__get_dir(), stdout=lf, stderr=lf)
 
+                # determine if the request is for a branch
+                is_branch = False
+                try:
+                    branch_listing = subprocess.check_output(['git', 'branch', '--all'], cwd=target_dir)
+                    branch_listing = branch_listing.replace('remotes/origin/', '')
+                    for branch in branch_listing.splitlines(True):
+                        if git_branch == branch.strip():
```
Comment (ptnapoleon) [inline ccmlib/repository.py]: It might be better to do `if git_branch in branch.strip():` here, just in case, no?

File: ccmlib/repository.py [hunk 3]
```diff
@@ -113,10 +113,32 @@ def clone_development(git_repo, version, verbose=False):
...
+                            is_branch = True
```
Comment (ptnapoleon) [inline ccmlib/repository.py]: Do we just want to break out of the for loop immediately if we find this?

File: ccmlib/repository.py [hunk 4]
```diff
@@ -113,10 +113,32 @@ def clone_development(git_repo, version, verbose=False):
...
+                except subprocess.CalledProcessError, cpe:
+                        print_("Error Running Branch Filter: %s\nAssumming request is not for a branch" % cpe.output)
+
                 # now check out the right version
                 if verbose:
-                    print_("Checking out requested branch (%s)" % git_branch)
-                out = subprocess.call(['git', 'checkout', git_branch], cwd=target_dir, stdout=lf, stderr=lf)
+                    branch_or_sha_tag = 'SHA/tag'
+                    if is_branch:
+                        branch_or_sha_tag = 'branch'
```
Comment (ptnapoleon) [inline ccmlib/repository.py]: You can condense these three lines down to:
`branch_or_sha_tag = 'branch' if is_branch else 'SHA/tag'`

**Comment by mambocab** [comment]
> Is this also valid python 3 syntax?

It's not. There's also some additional finagling required for Python 3 compat; I can take care of that.

File: ccmlib/repository.py (hunk 2 above)
Comment (mikefero) [inline ccmlib/repository.py]: That would probably be better than the replace for sure.

File: ccmlib/repository.py (hunk 3 above)
Comment (mikefero) [inline ccmlib/repository.py]: short circuit indeed

File: ccmlib/repository.py (hunk 4 above)
Comment (mikefero) [inline ccmlib/repository.py]: +1; didn't see any use of the ternary operation when I did a quick glance so I opted not to do that

**Pull request merged**